extensions/ql-vscode/src/helpers.ts

```diff
@@ -43,6 +44,13 @@ export const tmpDirDisposal = {
   },
 };
 
+interface ShowAndLogExceptionOptions extends ShowAndLogOptions {
+  /** Custom properties to include in the telemetry report. */
+  extraTelemetryProperties?: { [key: string]: string };
+  /** An alternate message to use for the notification instead of the message from the `Error`. */
+  notificationMessage?: string;
```

Comment:
Do you want to restrict this to a known set of strings? This would help with type checking and making sure you're not accidentally misspelling something. Eg-

```suggestion
  notificationMessage?: "preview_query_help" | "compare_view_show_results" | "eval_log_viewer_reveal" | ... and so on;
```

You could even extract this into its own type.

Reply:
This is the value that is shown to the user in the popup notification, so I don't think we can limit it to a fixed set of strings. It has to be able to contain custom data like database names or file paths. But this value isn't sent in the telemetry so it can safely contain any data.